### DISCUSSION TITLE
Validator custom messages change {{ field }} name

### DISCUSSION THREAD

**Discussion by jarstone** [adonisjs/core#3198] [Help]
For example I have a validation schema as follows:
```javascript
await request.validate({
  schema: schema.create({
    fullname: schema.string(),
    email: schema.string({}, [rules.email()]),
    // ...
  }),
  messages: {
    required: '{{ field }} is required.',
  },
})
```
If the "required" rule is not passed it will display the following messages:
```
fullname is required.
email is required.
```
How to replace field name with custom field name.
For example I want to replace "fullname" with "Full name" and "email" with "Email address".

Of course I can change it manually in the following way:
```javascript
messages: {
  'fullname.required': 'Full name is required.',
  'email.required': 'Email address is required.',
}
```

But I don't think it's an "elegant" way.
I migrated from Laravel to Adonis because I think the concept is similar. In laravel there is a way to do this, see https://laravel.com/docs/8.x/validation#customizing-the-validation-attributes

Is there some kind of way to do it in adonis without having to "hard coded" the message ?

**Comment by AleDiBernardo** [comment]
3 years later and i cannot find a simple way to do that, i also migrated from laravel

**Reply by thetutlage** [reply]
https://vinejs.dev/docs/custom_error_messages#field-names-substitution :)

**Reply by AleDiBernardo** [reply]
Can I create a separate file in the validators folder and then import it into the controllers where I validate the data with .validate(data,messages)?

**Reply by thetutlage** [reply]
Yes you can. You can create a message provider inside a file and export it and then within the controller you can say `.validate(data, { messagesProvider: importedMessagesProvider })`

**Reply by AleDiBernardo** [reply]
Thanks a lot
